autoload/airline/extensions/tabline/ctrlspace.vim

```diff
@@ -30,7 +30,15 @@ endfunction
 
 function! airline#extensions#tabline#ctrlspace#add_buffer_section(builder, cur_tab, cur_buf, pull_right)
   let pos_extension = (a:pull_right ? '_right' : '')
-  let buffer_list = ctrlspace#api#BufferList(a:cur_tab)
+
+  let l:buffer_list = []
+	for l:bufferindex in sort(keys(g:ctrlspace#api#Buffers(a:cur_tab), 'N'))
```

Comment:
why the for-loop? Can't you simply sort the resulting `ctrlspace#api#BufferList`?

Reply:
Well, the `ctrlspace#api#BufferList` return a list of dictionary like the following:
```vim
[{'index': 9, 'modified': 0, 'text': 'lists', 'visible': 0}, {'index': 10, 'modified': 0, 'text': 'memos.json', 'visible': 1}, {'index': 3, 'modified': 0, 'text': '.config/nvim/LICENSE', 'visible': 0}, {'index': 6, 'modified': 0, 'text': '.config/nvim/README.md', 'visible': 0}, {'index': 8, 'modified': 0, 'text': '.config/nvim/coc-settings.json', 'visible': 0}]
```
to sort this dictionary we will need to pass to the `sort()` function a function to sort this list of dictionaries depending on the value of 'index' key in each dictionary. And in general, calling function is more expensive than for loop. 

Also, of course there will be a slow down but this is not noticeable and we are trying to trade off between correct behavior and performance.

If there is any alternative method I can definitely help implement it